gk/main.c

```diff
@@ -584,15 +709,40 @@ alloc_gk_conf(void)
 }
 
 int
+gk_conf_hold(struct gk_config *gk_conf)
+{
+	if (gk_conf) {
```

Comment:
Why did you test `gk_conf`? I think you can drop this test, make the return void, and make the function inline.